### ISSUE TITLE
state_tax_by_gross_receipts implemented properly? and adding another modification to state income tax

### ISSUE THREAD

**Issue by daltonwstewart** [BioSTEAMDevelopmentGroup/BLocS#6]
Hi @yoelcortes 

While we're debugging, I wanted to make another change to how income tax is assessed for certain states, similar to [`state_tax_by_gross_receipts`](https://github.com/BioSTEAMDevelopmentGroup/BLocS/blob/main/blocs/incentives/evaluation.py#L203) we created before. However, I noticed that I couldn't actually find this function defined anywhere in `incentives_tea.py` or any of the biosteam or biorefineries TEA files. Could you take a look and make sure I didn't miss it anywhere? I can explain what the function should do if we need to redefine it.

Separately, the other change I wanted to make is that some states allow federal income tax to be subtracted from income before state income tax is assessed, so I would like to define a function that does the following (I just don't know the actual variable names or where to put it - I was going to model it after state_tax_by_gross_receipts when I realized I couldn't find the definition):
`def deduct_fed_inc_tax(): state_income_taxable_value = income_taxable_value - fed_inc_tax_assessed`

So if it isn't clear, we should determine federal income tax based on income and losses and charge that as a loss as normal, but also subtract this amount of tax as a loss before we assess state income tax. This is only allowed in 4 states. If you help me define the function I can set it up in `evaluation.py` with the proper if statement.

Thank you!

**Comment by yoelcortes** [comment]
Hi Dalton,

Did you know you can browse the files at point in history? Just go to "commits" on the landing page of this repository, then click on the "< >" button by the commit you would like to browse. I tried this for a couple of your commits in June, but could not find `state_tax_by_gross_receipts`. Do you know when you created it? Maybe its in the history.

Sure, I just implemented it. Just set the attribute `<TEA>.deduct_federal_income_tax_to_state_taxable_earnings` to True to deduct the federal income tax for the appropriate states.

Hope this helps,
Thanks!

**Comment by daltonwstewart** [comment]
Hi Yoel,

I did briefly look through some older commits, but I just wanted to check with you that I was looking in the right place. I eventually found the definition in the older repository that became BLocS, so it either wasn't copied over or was deleted somehow. Regardless, I added it to `incentives_tea.py` and I think both of the new functions are working properly. Thanks for your help.

Dalton
